src/object.d

```diff
@@ -285,8 +285,7 @@ class TypeInfo
 
     /// Return default initializer.  If the type should be initialized to all zeros,
     /// an array with a null ptr and a length equal to the type size will be returned.
-    // TODO: make this a property, but may need to be renamed to diambiguate with T.init...
-    const(void)[] init() nothrow pure const @safe @nogc { return null; }
+    abstract const(void)[] init() nothrow pure const @nogc;
```

Comment:
I think `@safe`, you can add a `@trusted` section to a safe function by using a function/delegate literal:

```D
override const(void)[] init() /* implied @safe */
{
   return (() @trusted => (cast(void *)null)[0..x])();
}
```

I don't like inferring `@trusted`, since it's very dangerous as an assumption. In fact, I'd say any virtual functions should never be marked `@trusted`.